src/mmw/js/src/modeling/models.js

```diff
@@ -517,10 +517,13 @@ var ScenariosCollection = Backbone.Collection.extend({
         });
 
         if (match) {
+            window.alert("There is another scenario with the same name. " +
+                    "Please choose a unique name for this scenario."
+            );
             console.log('This name is already in use.');
-            return;
+            return false;
         } else if (model.get('name') !== newName) {
-            model.set('name', newName);
+            return model.set('name', newName);
```

Comment:
If I don't need the returned value on the other side, I personally try to return boolean rather than truthy and falsey values.